api/peering_test.go

```diff
@@ -78,17 +216,19 @@ func TestAPI_Peering_GenerateToken_Read_Initiate_Delete(t *testing.T) {
 	// wait for the peering backend to finish the peering connection
 	time.Sleep(2 * time.Second)
 
-	respr, qm2, err4 := c2.Peerings().Read(ctx, "peer1", nil)
+	retry.Run(t, func(r *retry.R) {
+		respr, qm2, err4 := c2.Peerings().Read(ctx, PeeringRequest{Name: "peer1"}, nil)
 
-	// basic ok checking
-	require.NoError(t, err4)
-	require.NotEmpty(t, qm2)
+		// basic ok checking
+		require.NoError(t, err4)
+		require.NotEmpty(t, qm2)
 
-	// require that the peering state is not undefined
-	require.Equal(t, INITIAL, respr.State)
+		// require that the peering state is not undefined
+		require.Equal(t, INITIAL, respr.State)
```

Comment:
In the retry closure we should use `r` not `t` (or else go test will be very angry at you on failure)

Reply:
ah that's a great catch, thank you!